solidity/random-beacon/contracts/ReimbursmentPool.sol

```diff
@@ -0,0 +1,111 @@
+    /// @param spender Address where the reimbursment is sent.
+    function refund(uint256 gasSpent, address spender) external onlyAuthorized {
+        uint256 gasPrice = tx.gasprice < maxGasPrice
+            ? tx.gasprice
+            : maxGasPrice;
+        uint256 refundAmount = (gasSpent + staticGasCost) * gasPrice;
+
+        // solhint-disable-next-line avoid-low-level-calls
+        (bool sent, ) = spender.call{value: refundAmount}("");
+        require(sent, "Failed to refund Ether");
+    }
+
+    /// @notice Authorize a contract that can interact with this reimbursment pool.
+    ///         Can be authorized by the governance only.
+    /// @param _contract Authorized contract.
+    function authorize(address _contract) external onlyOwner {
+        isAuthorized[_contract] = true;
+    }
+
+    /// @notice Unauthorize a contract that was previously authorized to interact
+    ///         with this reimbursment pool. Can be unauthorized by the
+    ///         governance only.
+    /// @param _contract Authorized contract.
+    function unauthorize(address _contract) external onlyOwner {
+        delete isAuthorized[_contract];
+    }
+
+    /// @notice Setting a static gas cost for executing a transaction. Can be set
+    ///         by the governance only.
+    /// @param _staticGasCost Static gas cost.
+    function setStaticGasCost(uint256 _staticGasCost) external onlyOwner {
+        staticGasCost = _staticGasCost;
+    }
+
+    /// @notice Setting a max gas price for transactions. Can be set by the
+    ///         governance only.
+    /// @param _maxGasPrice Max gas price used to reimburse tx submitters.
+    function setMaxGasPrice(uint256 _maxGasPrice) external onlyOwner {
+        maxGasPrice = _maxGasPrice;
+    }
+
+    /// @notice Withdraws ETH amount from this pool which are sent to a given
+    ///         address. Can be set by the governance only.
+    /// @param amount Amount to withdraw from the pool.
+    /// @param to An address where ETH is sent.
+    function withdraw(uint256 amount, address to) external onlyOwner {
+        require(
+            address(this).balance >= amount,
+            "Insufficient contract balance"
+        );
+
+        // solhint-disable-next-line avoid-low-level-calls
+        (bool sent, ) = to.call{value: amount}("");
+        require(sent, "Failed to send Ether");
+    }
+
+    /// @notice Withdraws all ETH from this pool which are sent to a given
+    ///         address. Can be set by the governance only.
+    /// @param to An address where ETH is sent.
+    function withdrawAll(address to) external onlyOwner {
+        require(address(this).balance > 0, "Nothing to withdraw");
+
+        // solhint-disable-next-line avoid-low-level-calls
+        (bool sent, ) = to.call{value: address(this).balance}("");
+        require(sent, "Failed to send Ether");
+    }
+
+    /// @notice Receive ETH
+    receive() external payable {}
```

Comment:
Should we limit this to DAO only?

Reply:
IMHO it's good this way.

Reply:
If someone wants to donate, there is no reason for stopping them.